core/src/main/java/org/apache/iceberg/BaseChangelogContentScanTask.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ResidualEvaluator;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+abstract class BaseChangelogContentScanTask<ThisT, F extends ContentFile<F>>
+    extends BaseContentScanTask<ThisT, F>
+    implements ChangelogScanTask {
+
+  private final int changeOrdinal;
+  private final long commitSnapshotId;
+
+  BaseChangelogContentScanTask(int changeOrdinal, long commitSnapshotId, F file,
+                               String schemaString, String specString, ResidualEvaluator residuals) {
+    super(file, schemaString, specString, residuals);
+    this.changeOrdinal = changeOrdinal;
+    this.commitSnapshotId = commitSnapshotId;
+  }
+
+  @Override
+  public int changeOrdinal() {
+    return changeOrdinal;
+  }
+
+  @Override
+  public long commitSnapshotId() {
+    return commitSnapshotId;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("change_ordinal", changeOrdinal)
+        .add("commit_snapshot_id", commitSnapshotId)
+        .add("file", file().path())
```

Comment:
There seems to be some ways to have a method toStringHelper() at each level and use it for inheritance, do you think its worth it?  https://github.com/google/guava/issues/1239

Reply:
The string representation would include the actual class name. I thought that would be sufficient. I don't think we include the file path or location of deletes files right now. The rest is the same across changelog tasks.

Locations may have some sensitive data. I'd probably keep it as-is for now to match the existing behavior. We can definitely reconsider the behavior in the future.

Reply:
Hm , maybe I might have been not clear, so just to clarify it wasn't suggesting to make toString in the child clases and logging other fields like location, but rather to reduce the code by having a method toStringHelper() from the base class (BaseContentScanTask) and re-use it (where we already have log file, partition_data, and residual) ?

Reply:
Oh, okay, got it now. Let me explore this idea.

Reply:
I tried it locally. I couldn't control the order of fields in child classes and I couldn't benefit from it in split scan tasks as that one does not extend `BaseContentScanTask`. I'd just probably keep it as-is.

Reply:
Sounds good.